ipatests/test_integration/test_sssd.py

```diff
@@ -403,3 +405,186 @@ def test_trustdomain_disable_disables_subdomain(self):
+        client = self.clients[0]
+
+        username = "testuser001"
+        userpasswd = 'Secret123'
+
+        # add a user and set password
+        tasks.create_active_user(master, username, userpasswd)
+        tasks.kinit_admin(master)
+
+        pub_keys = []
+
+        privkey, pubkey = tasks.generate_ssh_keypair()
+        pub_keys.append(pubkey)
+        with open(os.path.join(
+                tmpdir, 'ssh_priv_key'), 'w') as fp:
+            fp.write(privkey)
+
+        master.run_command([
+            'ipa', 'user-mod', username, '--ssh', "{}".format(pub_keys[0])
+        ])
+
+        # connect with first SSH key
+        user_ssh_priv_key = os.path.join(tmpdir, 'ssh_priv_key')
+
+        ssh_key_file_contents = ''
+        with open(user_ssh_priv_key, 'r') as infile:
+            ssh_key_file_contents = infile.read()
+        master.put_file_contents('/tmp/user_ssh_priv_key',
+                                 ssh_key_file_contents)
+        master.run_command(['chmod', '600', '/tmp/user_ssh_priv_key'])
+
+        # add group a
+        cmd_output = master.run_command(['ipa', 'group-add', 'a'])
+        assert 'Added group "a"' in cmd_output.stdout_text
+
+        # add group b
+        cmd_output = master.run_command(['ipa', 'group-add', 'b'])
+        assert 'Added group "b"' in cmd_output.stdout_text
+
+        # add group c
+        cmd_output = master.run_command(['ipa', 'group-add', 'c'])
+        assert 'Added group "c"' in cmd_output.stdout_text
+
+        # add group members
+        cmd_output = master.run_command(['ipa', 'group-add-member',
+                                         'b', '--groups', 'a'])
+        assert 'Group name: b' in cmd_output.stdout_text
+        assert 'Member groups: a' in cmd_output.stdout_text
+        assert 'Number of members added 1' in cmd_output.stdout_text
+
+        cmd_output = master.run_command(['ipa', 'group-add-member',
+                                         'c', '--groups', 'b'])
+        assert 'Group name: c' in cmd_output.stdout_text
+        assert 'Member groups: b' in cmd_output.stdout_text
+        assert 'Indirect Member groups: a' in cmd_output.stdout_text
+
+        # add user to group 'a'
+        cmd_output = master.run_command(['ipa', 'group-add-member',
+                                         'a', '--users', username])
+        assert 'Group name: a' in cmd_output.stdout_text
+        assert 'Member users: {}'.format(username) in cmd_output.stdout_text
+        assert 'Member of groups: b' in cmd_output.stdout_text
+        assert 'Indirect Member of group: c' in cmd_output.stdout_text
+
+        # clear sssd_cache
+        clear_sssd_cache(master)
+
+        # user lookup
+        # at this point, testuser001 has the following group memberships
+        # Member of groups: a, ipausers
+        # Indirect Member of group: b, c
+        cmd_output = master.run_command(['ipa', 'user-show', username])
+        assert 'a' in cmd_output.stdout_text
```

Comment:
The assert is too vague. It will match for instance the line "First n**a**me: testuser001". I would also advise to name the groups groupa, groupb and groupc to avoid this type of issue instead of a,b,c.

Reply:
used groupa, groupb and groupc